sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java

```diff
@@ -16,7 +16,8 @@
  */
 public final class NettyAsyncHttpClientProvider implements HttpClientProvider {
     private static final int DEFAULT_MAX_CONNECTIONS = 500;
-    private static final AtomicReference<HttpClient> DEFAULT_HTTP_CLIENT = new AtomicReference<>();
+    private static final AtomicReference<HttpClient> DEFAULT_HTTP_CLIENT =
+        new AtomicReference<>(new NettyAsyncHttpClientBuilder().build());
```

Comment:
If we use static initializer for this then AtomicReference isn't necessary (as long as field is immutable, aka final).

I'm not sure though if this is right pattern given that some clients spin up internal resources. If this for any reason isn't used it's a waste.

Please check the "double checked locking" to implement lazy singleton (see https://www.baeldung.com/java-singleton-double-checked-locking ).